src/codewhisperer/vue/backend.ts

```diff
@@ -0,0 +1,153 @@
+                })
+            })
+        } else {
+            this.saveFileLocally(localFilePath, fileContent)
+        }
+    }
+
+    // This function saves and open the file in the editor.
+    private async saveFileLocally(localFilePath: string, fileContent: string): Promise<void> {
+        await fs.promises.writeFile(localFilePath, fileContent)
+        this.isFileSaved = true
+        vscode.workspace.openTextDocument(localFilePath).then(doc => {
+            vscode.window.showTextDocument(doc, vscode.ViewColumn.Active).then(editor => {
+                const endOfDocument = new vscode.Position(doc.lineCount - 1, doc.lineAt(doc.lineCount - 1).text.length)
+                editor.selection = new vscode.Selection(endOfDocument, endOfDocument)
+            })
+        })
+    }
+
+    //This function returns the OS type of the machine used in Shortcuts and Generate Suggestion Sections
+    public getOSType() {
+        const platform = os.platform()
+        if (platform === 'darwin') {
+            return 'Mac'
+        } else {
+            return 'RestOfOS'
+        }
+    }
+
+    //This function opens the Keyboard shortcuts in VSCode
+    async openShortCuts(): Promise<void> {
+        vscode.commands.executeCommand('workbench.action.openGlobalKeybindings', 'codewhisperer')
+    }
+
+    //This function opens the Feedback CodeWhisperer page in the webview
+    async openFeedBack(): Promise<void> {
+        vscode.commands.executeCommand('aws.submitFeedback', 'CodeWhisperer')
+    }
+
+    //------Telemetry------
+    /** This represents the cause for the webview to open, whether a certain button was clicked or it opened automatically */
+    #codeWhispererSource?: CodeWhispererSource
+
+    setSource(source: CodeWhispererSource | undefined) {
+        if (this.#codeWhispererSource) {
+            return
+        }
+        this.#codeWhispererSource = source
+    }
+
+    emitUiClick(id: CodeWhispererUiClick) {
+        telemetry.ui_click.emit({
+            elementId: id,
+        })
+    }
+    //Telemetry for CodeWhisperer Try Example with two params Language and Task Type
+    emitTryExampleClick(languageSelected: CodewhispererLanguage, taskType: CodewhispererGettingStartedTask) {
+        telemetry.codewhisperer_onboardingClick.emit({
+            codewhispererLanguage: languageSelected,
+            codewhispererGettingStartedTask: taskType,
+        })
+    }
+}
+// export type codewhispererLanguage = 'python' | 'javascript' | 'csharp'
+//List of all events that are emitted from the webview of CodeWhisperer
+export type CodeWhispererUiClick =
+    | 'codeWhisperer_Resources_Documentation'
+    | 'codeWhisperer_Resources_Feedback'
+    | 'codeWhisperer_Workshop_LearnMore'
+    | 'codeWhisperer_Commands_KeyboardShortcutsEditor'
+    | 'codeWhisperer_ScanCode_LearnMore'
+    | 'codeWhisperer_GenerateSuggestions_LearnMore'
+    | 'codeWhisperer_GenerateSuggestions_TryExample'
```

Comment:
Where do we capture the page open click event? I don't see that
`codeWhisperer_GettingStarted_PageOpen`

Reply:
do you also want to put `codewhisperer_learn_buttonclick` here?

Reply:
`codeWhisperer_GettingStarted_PageOpen`

That was in other branch as that requires additional files from **aws:Master**
Which was not part of my PR.
I will raise a separate PR for the Telemetry of `codeWhisperer_GettingStarted_PageOpen` after this PR was resolved

Reply:
> do you also want to put `codewhisperer_learn_buttonclick` here? And also to make it consistent `codeWhisperer_Learn_ButtonClick`

Updated accordingly.

Reply:
> do you also want to put `codewhisperer_learn_buttonclick` here? And also to make it consistent `codeWhisperer_Learn_ButtonClick`

We can not use here as `Learn` button is not a part of CodeWhispererWebView.